app/controllers/api/v1/admin/trends/tags_controller.rb

```diff
@@ -1,16 +1,39 @@
 # frozen_string_literal: true
 
 class Api::V1::Admin::Trends::TagsController < Api::V1::Trends::TagsController
-  before_action -> { authorize_if_got_token! :'admin:read' }
+  include Authorization
+
+  before_action -> { authorize_if_got_token! :'admin:read' }, only: :index
+  before_action -> { authorize_if_got_token! :'admin:write' }, except: :index
+
+  after_action :verify_authorized
 
   def index
+    authorize :tag, :index?
```

Comment:
Same remark as with `Api::V1::Admin::Trends::LinksController` with regards to changing the endpoint's behavior. In particular, this will break hashtags in the admin dashboard for moderators that do not have the `manage_taxonomies` permission.

Reply:
You're right. I got a little overzealous writing the tests for these new methods. I have fixed it.